src/hotspot/share/classfile/classLoaderExt.cpp

```diff
@@ -260,47 +260,32 @@ InstanceKlass* ClassLoaderExt::load_class(Symbol* name, const char* path, TRAPS)
   assert(DumpSharedSpaces, "this function is only used with -Xshare:dump");
   ResourceMark rm(THREAD);
   const char* class_name = name->as_C_string();
-
   const char* file_name = file_name_for_class_name(class_name,
                                                    name->utf8_length());
   assert(file_name != NULL, "invariant");
 
   // Lookup stream for parsing .class file
   ClassFileStream* stream = NULL;
-  ClassPathEntry* e = find_classpath_entry_from_cache(path, CHECK_NULL);
+  ClassPathEntry* e = find_classpath_entry_from_cache(THREAD, path);
   if (e == NULL) {
-    return NULL;
-  }
-  {
-    PerfClassTraceTime vmtimer(perf_sys_class_lookup_time(),
-                               THREAD->as_Java_thread()->get_thread_stat()->perf_timers_addr(),
-                               PerfClassTraceTime::CLASS_LOAD);
-    stream = e->open_stream(file_name, CHECK_NULL);
+    THROW_NULL(vmSymbols::java_lang_ClassNotFoundException());
```

Comment:
Again not obvious this is the right exception to throw. It is also not very informative as to what the actual problem was - isn't it the "source:" that was not found here?

Reply:
This will be improved in [JDK-8263469](https://bugs.openjdk.java.net/browse/JDK-8263469).